src/descriptor/policy.rs

```diff
@@ -911,6 +925,12 @@ impl<Ctx: ScriptContext> ExtractPolicy for Miniscript<DescriptorPublicKey, Ctx>
     }
 }
 
+fn psbt_inputs_sat(psbt: &PSBT) -> Vec<PsbtInputSatisfier> {
+    (0..psbt.inputs.len())
+        .map(|i| PsbtInputSatisfier::new(psbt, i))
+        .collect()
```

Comment:
I think you can avoid allocating a vec here if you just return `impl Iterator<Item=PsbtInputSatisfier>`. You don't really need the vector anyway

Reply:
should be fixed in 136a4bddb2571a3db3994c2bf29f715ba66a70b8